cpp/src/arrow/pretty_print.cc

```diff
@@ -670,4 +670,49 @@ Status PrettyPrint(const Schema& schema, const PrettyPrintOptions& options,
   return Status::OK();
 }
 
+void GdbPrintArray(const Array& arr, int indent) {
```

Comment:
These changes aren't related to the PR and I can pull them out but I'd like to get them in at some point.  The GDB pretty prints (python scripts) aren't included in this PR but I use them for debugging and they rely on these functions.

Reply:
Thanks for working on these, but please revert them here and add them in a dedicated PR

Reply:
Removed.